File: src/main/java/com/elissamc/games/GameHandler.java
package com.elissamc.games;

import cn.nukkit.Player;
import cn.nukkit.Server;
import cn.nukkit.event.Listener;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public class GameHandler implements Listener {

    private final Map<String, GameParent> games = new HashMap<>();
    private final Map<UUID, String> playerStatus = new HashMap<>();

    public boolean joinGame(Player player, String s) {
        if (games.containsKey(s)) {
            games.get(s).addPlayer(player);
            playerStatus.put(player.getUniqueId(), s);
            return true;
        }
        return false;
    }

    public void addGame(String name, GameParent lobbys) {
        games.put(name, lobbys);
    }

    public void removeGame(Player player) {
        if (playerStatus.containsKey(player.getUniqueId())) {
            games.get(playerStatus.get(player.getUniqueId())).removePlayer(player);
            player.teleport(Server.getInstance().getDefaultLevel().getSafeSpawn());
        }
    }
}

File: src/main/java/com/elissamc/games/GameHandler.kt
package com.elissamc.games

import cn.nukkit.Player
import cn.nukkit.Server
import cn.nukkit.event.Listener

import java.util.HashMap
import java.util.UUID

class GameHandler : Listener {

    val games = HashMap<String, GameParent>()
    private val playerStatus = HashMap<UUID, String>()

    fun getGames(): Map<String, GameParent> {
        return games
    }

    fun joinGame(player: Player, s: String): Boolean {
        if (games.containsKey(s)) {
            games[s]!!.addPlayer(player)
            playerStatus[player.uniqueId] = s
            return true
        }
        return false
    }

    fun addGame(name: String, lobbys: GameParent) {
        games[name] = lobbys
    }

    fun removeGame(player: Player) {
        if (playerStatus.containsKey(player.uniqueId)) {
            games[playerStatus[player.uniqueId]]!!.removePlayer(player)
            player.teleport(Server.getInstance().defaultLevel.safeSpawn)
        }
    }
}